pkg/productinfo/types.go

```diff
@@ -166,9 +166,11 @@ type ProductDetailSource interface {
 }
 
 // newProductDetails creates a new ProductDetails struct and returns a pointer to it
-func newProductDetails(vm VmInfo) *ProductDetails {
+func (cpi *CachingProductInfo) newProductDetails(vm VmInfo, provider string) *ProductDetails {
```

Comment:
please change the order of the arguments; move this method in the file where the other methods of the cpi are implemented